File: src/main/java/no/nav/permitteringsskjemaapi/replay/ReplayRepository.java
package no.nav.permitteringsskjemaapi.replay;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;
import org.springframework.data.jpa.repository.Query;

import javax.persistence.LockModeType;
import java.util.List;
import java.util.UUID;

public interface ReplayRepository extends JpaRepository<ReplayQueueItem, UUID> {

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    @Query(
            value = "select * from replay_queue_items limit 10",
            nativeQuery = true
    )
    List<ReplayQueueItem> fetchReplayQueueItems();
}

File: src/main/kotlin/no/nav/permitteringsskjemaapi/replay/ReplayRepository.kt
package no.nav.permitteringsskjemaapi.replay

import jakarta.persistence.LockModeType
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import org.springframework.data.jpa.repository.Query
import java.util.*

interface ReplayRepository : JpaRepository<ReplayQueueItem, UUID> {
    @Lock(LockModeType.PESSIMISTIC_WRITE)
    @Query(value = "select r from ReplayQueueItem r")
    fun fetchReplayQueueItems(pageable: Pageable): List<ReplayQueueItem>
}